File: frontend/src/app/services/userViewModel.service.ts
import {Injectable} from "@angular/core";
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {UserViewModel} from "../models/userViewModel";
import {User} from "../models/userModel";

@Injectable()
export class UserViewModelService {
  constructor(private http: HttpClient) {

  }

  getUserViewModel(id: number) : Observable<UserViewModel> {
    return this.http.get<UserViewModel>('/api/userview/' + id);
  }

}

File: fapi/src/main/java/com/netcracker/ivanmerkush/fapi/service/UserViewModelService.java
package com.netcracker.ivanmerkush.fapi.service;

import com.netcracker.ivanmerkush.fapi.models.User;
import com.netcracker.ivanmerkush.fapi.models.UserViewModel;

public interface UserViewModelService {
    UserViewModel getUserViewModel(Integer id);
}
